Mailbox/DataLoader.cs

```diff
@@ -1,21 +1,91 @@
-﻿using System.Collections.Generic;
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
 using System.IO;
 
 namespace Mailbox
 {
-    public class DataLoader
+    public class DataLoader : IDisposable
     {
+
+        private Stream Source { get; }
         public DataLoader(Stream source)
         {
+            Source = source ?? throw new ArgumentNullException(nameof(source));
         }
 
         public List<Mailbox> Load()
         {
+            List<Mailbox> mailboxes = new List<Mailbox>();
+            Source.Position = 0;
+
+            try
+            {
+                using(StreamReader reader = new StreamReader(Source, leaveOpen:true))
+                {
+                    string line;
+                    while((line = reader.ReadLine()) != null){
+                        Mailbox mailBox = JsonConvert.DeserializeObject<Mailbox>(line);
+                        mailboxes.Add(mailBox);
+                    }
+                }
+            }
+            catch (JsonReaderException)
+            {
+                return null;
+            }
+
+            return mailboxes;
+        }
+
+        public void Save(List<Mailbox> mailboxes) //TODO
+        {
+            Source.Position = 0;
+
+            using(StreamWriter writer = new StreamWriter(Source, leaveOpen:true))
+            {
+                foreach (Mailbox mB in mailboxes)
+                {
+                    string json = JsonConvert.SerializeObject(mB);
+                    writer.WriteLine(json);
+                }
+            }
+        }
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    // TODO: dispose managed state (managed objects).
+                    Source?.Dispose();
```

Comment:
Since you null checked `Source` on the way in and it is a readonly on property. You don't need the null check here.
```suggestion
                    Source.Dispose();
```